Desktop/widgets/variableslistbase.cpp

```diff
@@ -306,6 +310,70 @@ void VariablesListBase::moveItems(QList<int> &indexes, ListModelDraggable* targe
 
 void VariablesListBase::termsChangedHandler()
 {
+	setColumnsTypes(model()->termsTypes());
+
 	if (_boundControl)	_boundControl->updateOption();
 	else JASPListControl::termsChangedHandler();
 }
+
+
+int VariablesListBase::_getAllowedColumnsTypes()
+{
+	int allowedColumnsTypes = -1;
+
+	if (!allowedColumns().isEmpty())
+	{
+		allowedColumnsTypes = 0;
+		for (QString& allowedColumn: allowedColumns())
+		{
+			columnType allowedType = columnTypeFromQString(allowedColumn, columnType::unknown);
+			if (allowedType != columnType::unknown)
+				allowedColumnsTypes |= int(allowedType);
+			else
+				addControlError(tr("Wrong column type: %1 for ListView %2").arg(allowedColumn).arg(name()));
+		}
+	}
+
+	return allowedColumnsTypes;
+}
+
+void VariablesListBase::_setAllowedVariables()
+{
+	if (suggestedColumns().empty() && !allowedColumns().empty())
+		setSuggestedColumns(allowedColumns());
+	else if (allowedColumns().empty() && !suggestedColumns().empty())
+	{
+		QStringList newAllowedColumns = suggestedColumns();
+		if (suggestedColumns().contains("scale"))
+		{
```

Comment:
I think it looks nicer as:
```
			if (!newAllowedColumns.contains("nominal")) newAllowedColumns.push_back("nominal");
			if (!newAllowedColumns.contains("ordinal"))   newAllowedColumns.push_back("ordinal");
```

But even better would be to turn the list into a set and simply do: `newAllowedColumns.insert("nominal")` 
Because there really is not reason to keep order for a list of allowed columns is there?